Assets/Scripts/Minigames/CowardMiniGame/RipcordBehavior.cs

```diff
@@ -12,64 +12,89 @@
 using TMPro;
 using UnityEngine.UI;
 using PlaceboEntertainment.UI;
+using UnityEngine.InputSystem;
+using System;
 
 public class RipcordBehavior : MonoBehaviour
 {
-    [SerializeField] private float _returnSpeed;
     [SerializeField] private GameObject _gears;
 
-    [Header("Scoring Information")]
-    [SerializeField] private GameObject _goalObject;
-    [SerializeField] private float _distanceFromGoalToScore;
+    [Header("Ripcord Stats")]
+    [SerializeField] private float _followPlayerSpeed;
+    [SerializeField] private Transform _maxPlayerDetectionPosition;
+    [SerializeField] private float _returnSpeed;
+    
+
+    [Header("Goal/Scoring Information")]
+    [SerializeField] private GameObject _scoringRangeObject;
+    [SerializeField] private Transform _maxGoalSpawnPosition;
 
     [Header("UI")]
     [SerializeField] private TextMeshPro _successfulPulls;
 
     [Header("VFX")]
     [SerializeField] private ParticleSystem _steam;
 
-    [Header("Lights")]
-    [SerializeField] private float _lightBlinkSpeed;
-    [SerializeField] private GameObject[] _lights;
-
     private PlayerController _playerController;
     private GameObject _player;
+
+    private Vector3 _startPosition;
+    private int _score;
+    private GameObject _scoreDetectionRange;
+
+    //bitmask to hit later 8 (player)
+    int layerMask = 1 << 8;
+
     private bool _isFollowingPlayer;
     private bool _lightIsBlinking;
     private bool _canScore;
     private bool _isMinigameActive;
-
-    private Vector3 _startPosition;
-
     private bool _isAtStartPosition;
-    private int _score;
-    private GameObject _goal;
 
-    // Start is called before the first frame update
+    public static Action OnRipcordScore;
+    public static Action<bool> OnPlayerDetectionChange;
+
     void Start()
     {
         _playerController = FindObjectOfType<PlayerController>();
         _player = _playerController.gameObject;
-        
         _isFollowingPlayer = false;
         _canScore = false;
         _isAtStartPosition = true;
         _score = 0;
         _startPosition = transform.position;
     }
 
+    private void FixedUpdate()
+    {
+        // Cancels the ripcord following if the player is out of the detection range
+        if (_isFollowingPlayer && Physics.OverlapCapsule(transform.position, _maxPlayerDetectionPosition.position, 1f, layerMask).Length <= 0)
```

Comment:
is this better than ontriggerexit? just asking for future reference

Reply:
Good question!
The range for the ripcord to detect if it can follow the player is different than the range for the player to interact with it. That means we would need two colliders, which is kinda messy, so I went with an overlap capsule to give us a large and more consistent detection range than a traditional raycast.